lib/master.js

```diff
@@ -246,7 +246,7 @@ Master.prototype._startWorkers = function(remainingWorkers, res) {
                 // Let all the exit listeners fire before reassigning current worker ID
                 process.nextTick(function() {
                     self._currentStartingWorker = undefined;
-                    resolve(self._startWorkers(remainingWorkers));
+                    resolve(self._startWorkers(remainingWorkers, res));
```

Comment:
I don't think so; this is inside an event handler that reacts to a worker exiting, so something that normally happens after startup. Workers are normally restarted until they successfully start up & send a return message, which will be returned after all workers have successfully started up.